### PR TITLE
fix(es/transforms/module): fix helper injection detection across import order

### LINKED ISSUE #2856: `_interopRequireWildcard` is injected without definition
### Describe the bug

When `import React from 'react'` is used, and also a `{...props}` with a `key={}` in JSX, `_interopRequireWildcard` is injected at the top of the TSX file while not being initialized. 

This create the error `ReferenceError: _interopRequireWildcard is not defined`

### Input code

```typescript
import React from 'react';

export const element = (
  <div {...innerTbodyProps} key={true ? 1 : 2} />
);
```

### Config

For TS:

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true
    },
    "transform": {
      "react": {
        "runtime": "automatic"
      }
    }
  },
  "module": {
    "type": "commonjs"
  }
}
```

For JS:

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": true
    },
    "transform": {
      "react": {
        "runtime": "automatic"
      }
    }
  },
  "module": {
    "type": "commonjs"
  }
}
```

Both have the same issue

### Output code

```tsx
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.element = void 0;
var _react = _interopRequireWildcard(require("react"));
function _defineProperty(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _objectSpread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {
        };
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _defineProperty(target, key, source[key]);
        });
    }
    return target;
}
const element = /*#__PURE__*/ (0, _react).createElement("div", _objectSpread({
}, innerTbodyProps, {
    key: true ? 1 : 2
}));
exports.element = element;
```

### Playground link

_No response_

### Expected behavior

it should haved inject `_interopRequireWildcard`, just like if `import * as React from 'react'` was used, so something like that:

```tsx
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.element = void 0;
var React = _interopRequireWildcard(require("react"));
function _defineProperty(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _interopRequireWildcard(obj) {
    if (obj && obj.__esModule) {
        return obj;
    } else {
        var newObj = {
        };
        if (obj != null) {
            for(var key in obj){
                if (Object.prototype.hasOwnProperty.call(obj, key)) {
                    var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {
                    };
                    if (desc.get || desc.set) {
                        Object.defineProperty(newObj, key, desc);
                    } else {
                        newObj[key] = obj[key];
                    }
                }
            }
        }
        newObj.default = obj;
        return newObj;
    }
}
function _objectSpread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {
        };
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _defineProperty(target, key, source[key]);
        });
    }
    return target;
}
const element = /*#__PURE__*/ (0, React).createElement("div", _objectSpread({
}, innerTbodyProps, {
    key: true ? 1 : 2
}));
exports.element = element;
```

### Version

1.2.112

### Additional context

I tried to use the swc playground to share a link, but it wasn't working. So here is a screenshot proving what I'm saying:

image

### PR DESCRIPTION
**Pull request by kwonoj**
- closes #2856

As commented in issue, crux is depends on import order import_analysis does not flag wildcard interop need to be used correctly. PR includes a scoped property to contain import sets to determine which interop is requred regardless of import order.

### CODE REVIEW

**Review by kdy1 — CHANGES_REQUESTED** [review]
Sorry, I reviewed this, but forgot submitting it.

**Comment by kwonoj** [comment]
> Sorry, I reviewed this, but forgot submitting it.

No worries, it's holiday around here and I'm slowing down as well.

File: crates/swc_ecma_transforms_module/src/import_analysis.rs [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
-    as_folder(ImportAnalyzer { scope })
+    as_folder(ImportAnalyzer {
+        scope,
+        import_srcs: Default::default(),
+    })
 }
 
 pub struct ImportAnalyzer {
     scope: Rc<RefCell<Scope>>,
+    import_srcs: AHashSet<JsWord>,
```
Comment (kwonoj) [inline crates/swc_ecma_transforms_module/src/import_analysis.rs:20]: Guess this is it?

Comment (kdy1) [reply crates/swc_ecma_transforms_module/src/import_analysis.rs:20]: Yes, correct.

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**